### PR TITLE
Add `npm version` pre and post tasks for release

### PR DESCRIPTION
**Pull request by lumberj**
Use `npm version {patch|minor|major}` to release new version

https://docs.npmjs.com/cli/version

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -24,7 +24,9 @@
     "test": "istanbul test -- _mocha",
     "coveralls": "cat ./coverage/lcov.info | ./node_modules/.bin/coveralls",
     "apidoc": "apidoc -o apidoc-out -i src/controllers/",
-    "integration": "integration all"
+    "integration": "integration all",
+    "preversion": "npm test",
+    "postversion": "git push --follow-tags"
```
Comment (clark800) [inline package.json]: Is it guaranteed to be published before the tag is pushed?

Comment (lumberj) [inline package.json]: Do you mean that `npm version` should also run `npm publish`? It does not by default

Comment (lumberj) [inline package.json]: The publish step is done by CI, so it's guaranteed that the tag is pushed first

**Comment by gip** [comment]
LGTM. Very cool feature.

**Pull request merged**